server/src/test/java/org/opensearch/rest/action/admin/cluster/RestClusterStatsActionTests.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.rest.FakeRestRequest;
+
+import java.util.HashMap;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.object.HasToString.hasToString;
+import static org.mockito.Mockito.mock;
+
+public class RestClusterStatsActionTests extends OpenSearchTestCase {
+
+    private RestClusterStatsAction action;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        action = new RestClusterStatsAction();
+    }
+
+    public void testUnrecognizedMetric() {
+        final HashMap<String, String> params = new HashMap<>();
+        final String metric = randomAlphaOfLength(64);
+        params.put("metric", metric);
+        final RestRequest request = new FakeRestRequest.Builder(xContentRegistry()).withPath("/_cluster/stats").withParams(params).build();
+        final IllegalArgumentException e = expectThrows(
+            IllegalArgumentException.class,
+            () -> action.prepareRequest(request, mock(NodeClient.class))
+        );
+        assertThat(e, hasToString(containsString("request [/_cluster/stats] contains unrecognized metric: [" + metric + "]")));
+    }
+
+    public void testUnrecognizedIndexMetric() {
```

Comment:
Could we test empty `index_metric` path?
`/_cluster/stats/_all//nodes/_all` ?

Reply:
This is a valid case, the tests in these files are for failure cases.

Reply:
What is the expected output in this case ?

Reply:
This will be treated as /_cluster/stats/_all/_all/nodes/_all, hence all stats will be returned.